### PR TITLE
Add generated types exports

### PR DESCRIPTION
**Pull request by justinkchen**
Update artifacts, add generated types from abi-gen to package exports, fix up tsconfig, fixed the contract generator template

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,14 +2,18 @@
   "compilerOptions": {
     "outDir": "./transpiled/",
     "sourceMap": true,
-    "noImplicitAny": true,
+    "noImplicitAny": false,
```
Comment (felix2feng) [inline tsconfig.json]: Has this been an issue with any dependencies?

File: utils/Web3Utils.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
+/*
```
Comment (felix2feng) [inline utils/Web3Utils.ts]: This seems to be a good candidate to put in the utils repo? where did this come from?

File: types/base_contract.ts [hunk 3]
```diff
@@ -35,7 +42,7 @@ export class BaseContract {
         };
         if (_.isUndefined(txDataWithDefaults.gas) && !_.isUndefined(estimateGasAsync)) {
             const estimatedGas = await estimateGasAsync(txData);
-            txDataWithDefaults.gas = estimatedGas;
+            txDataWithDefaults.gas = new BigNumber(estimatedGas);
```
Comment (felix2feng) [inline types/base_contract.ts]: What was the need for the change here?

**Comment by felix2feng** [comment]
😢 Coverage. Error is:
```
Error: Cannot find module '../../artifacts/ts/Authorizable'
```

File: types/base_contract.ts (hunk 3 above)
Comment (justinkchen) [reply types/base_contract.ts]: this is just what was being done in the setprotocol.js side so decided to do it here

File: utils/Web3Utils.ts (hunk 2 above)
Comment (justinkchen) [reply utils/Web3Utils.ts]: we'be been using it in both sides but its based off of the web3-utils package

File: tsconfig.json (hunk 1 above)
Comment (justinkchen) [reply tsconfig.json]: causes a lot of issues with the artifacts that are autogenerated

**Pull request merged**